validation/views.py

```diff
@@ -59,7 +68,92 @@ def search_phrases(request):
     """
     The search results for pages.
     """
-    return HttpResponse(501)
+    query = request.GET.get("query")
+    cree_matches = Phrase.objects.filter(transcription__contains=query)
+    english_matches = Phrase.objects.filter(translation__contains=query)
+    all_matches = list(set().union(cree_matches, english_matches))
+    paginator = Paginator(all_matches, 30)
+    page_no = request.GET.get("page", 1)
+    phrases = paginator.get_page(page_no)
+    context = dict(phrases=phrases, search_term=query)
+    return render(request, "validation/search.html", context)
+
+
+def advanced_search(request):
+    """
+    The search results for pages.
+    """
+    query = Speaker.objects.all()
+    speakers = []
+    for q in query:
+        speakers.append(q.code)
+
+    context = dict(speakers=speakers)
+    return render(request, "validation/advanced_search.html", context)
+
+
+def advanced_search_results(request):
+    """
+    Search results for advanced search
+    Takes in parameters from GET request and makes necessary
+    queries to return all the appropriate entries
+    The steps are:
+    cree phrases UNION english phrases UNION analysis
+    INTERSECT status
+    INTERSECT speaker
+    """
+    transcription = request.GET.get("transcription")
+    translation = request.GET.get("translation")
+    analysis = request.GET.get("analysis")
+    status = request.GET.get("status")
+    speaker = request.GET.get("speaker")
+    speakers = speaker.strip().split(",")[:-1]
```

Comment:
Can replace with [`QueryDict.getlist()`](https://docs.djangoproject.com/en/3.1/ref/request-response/#django.http.QueryDict.getlist):

```suggestion
    speakers = request.GET.getlist("speaker-options")
```

(see above JavaScript comments)